### PR TITLE
feat(keyboard): Accept codes

### PR DESCRIPTION
**Pull request by JoelEinbinder**
This patch lets key names be [code](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/code) in addition to [key](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key). When specifying a code, the proper text is generated assuming a standard US keyboard layout. e.g `Digit5` -> `"5"` or `"%"` depending on `Shift`. 

[location](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/location) is now specified.

Using unknown key names now throws an error.

Typing newlines now correctly presses enter.

### CODE REVIEW

File: lib/Input.js [hunk 1]
```diff
@@ -30,23 +40,25 @@ class Keyboard {
...
-  async down(key, options = {text: undefined}) {
-    let { text } = options;
-    // If the key is a single character, and no modifiers are pressed except shift, a keypress should be generated by default.
-    if (text === undefined)
-      text = (key.length === 1 && !(this._modifiers & ~8)) ? key : '';
-    const autoRepeat = this._pressedKeys.has(key);
-    this._pressedKeys.add(key);
-    this._modifiers |= this._modifierBit(key);
+  async down(key, options = { text: undefined }) {
+    const description = this._keyDescriptionForString(key);
```
Comment (aslushnikov) [inline lib/Input.js:44]: this should throw if there's no description for the key

File: lib/Input.js [hunk 2]
```diff
@@ -67,17 +79,71 @@ class Keyboard {
...
+    return description;
+  }
+
+  /**
    * @param {string} key
    */
   async up(key) {
-    this._modifiers &= ~this._modifierBit(key);
-    this._pressedKeys.delete(key);
+    const description = this._keyDescriptionForString(key);
```
Comment (aslushnikov) [inline lib/Input.js:133]: this throws if there's no key description as well

**Review by aslushnikov — APPROVED** [review]

File: lib/Input.js (hunk 1 above)
Comment (JoelEinbinder) [reply lib/Input.js:44]: Done.

File: lib/Input.js (hunk 2 above)
Comment (JoelEinbinder) [reply lib/Input.js:133]: Done.

**Comment by aslushnikov** [comment]
let's at least reference bugs this is about.

**Comment by JoelEinbinder** [comment]
This should help/fix
#681 
#777
#723

**Pull request merged**